packages/react-storage/src/components/StorageBrowser/context/locationActions/types.ts

```diff
@@ -33,7 +33,7 @@ export interface LocationActions<T = Permission> {
 
 export type LocationActionsAction<T = string> =
   | { type: 'CLEAR' }
-  | { type: 'SET_ACTION'; payload: T }
+  | { type: 'SET_ACTION'; payload: T; files?: File[] }
```

Comment:
Adding `files` to SET_ACTION for the purpose of carrying files dropped on the Location Detail view into the Upload Action view

Reply:
`payload` only makes sense here if it encompasses all provided data of the action. Should either pass both `type` and `file` as `payload` or change `payload` to `actionType`

Reply:
Changed to `actionType`